supervisor/addons/addon.py

```diff
@@ -608,6 +609,11 @@ async def write_options(self) -> None:
 
     async def unload(self) -> None:
         """Unload add-on and remove data."""
+        if self._startup_task:
+            # If we were waiting on startup, cancel that and let the task finish before proceeding
+            self._startup_task.cancel(f"Removing add-on {self.name} from system")
+            await asyncio.sleep(0)
```

Comment:
This seems a bit ad-hoc way to let the event loop to run.

Couldn't we just await the task here?

Reply:
No because that's not the task we're letting run. Something else is awaiting the result of that task, like the API call the user made or the backup/restore the user started. That's what we want to let run. Otherwise if we remove the addons data that task may get errors trying to access its name or something and expecting that to be there.